### ISSUE TITLE
AttributeError: 'CLIPTextModel' object has no attribute 'peft_config' when using default train_dreambooth_lora_sdxl_advanced.py but with checkpointing_steps < max_train_steps

### ISSUE THREAD

**Issue by brandostrong** [huggingface/diffusers#6444]
### Describe the bug

When checkpointing at defined steps, such as every 50 steps I get a 'CLIPTextModel' object has no attribute 'peft_config'  while trying to save the model. This only happens when checkpointing_steps < max_train_steps, as the code wouldn't run if it was higher than the train steps

### Reproduction

python train_dreambooth_lora_sdxl_advanced.py --pretrained_model_name_or_path="stabilityai/stable-diffusion-xl-base-1.0" --pretrained_vae_model_name_or_path="madebyollin/sdxl-vae-fp16-fix" --dataset_name="./trainimages/" --caption_column=prompt --instance_prompt="a TOK person" --validation_prompt="a TOK person dressed as Yoda" --output_dir="huggy_v11" --caption_column="prompt" --mixed_precision="bf16" --resolution=1024 --train_batch_size=1 --repeats=1 --gradient_accumulation_steps=1 --gradient_checkpointing --train_text_encoder_ti --learning_rate=1 --text_encoder_lr=1 --lr_scheduler="constant" --snr_gamma=5.0 --lr_warmup_steps=0 --rank=32 --max_train_steps=1000 --checkpointing_steps=50 --seed="0" --optimizer="prodigy" --prodigy_safeguard_warmup=True --prodigy_use_bias_correction=True --adam_beta1=0.9 --adam_beta2=0.99 --adam_weight_decay=0.01

### Logs

```shell
Traceback (most recent call last):
  File "C:\Users\\Documents\GitHub\\train_dreambooth_lora_sdxl_advanced.py", line 2100, in <module>
    main(args)
  File "C:\Users\\Documents\GitHub\\train_dreambooth_lora_sdxl_advanced.py", line 1895, in main
    accelerator.save_state(save_path)
  File "C:\Users\\Documents\GitHub\\venv\lib\site-packages\accelerate\accelerator.py", line 2793, in save_state
    hook(self._models, weights, output_dir)
  File "C:\Users\\Documents\GitHub\\train_dreambooth_lora_sdxl_advanced.py", line 1320, in save_model_hook
    get_peft_model_state_dict(model)
  File "C:\Users\\Documents\GitHub\\venv\lib\site-packages\peft\utils\save_and_load.py", line 64, in get_peft_model_state_dict
    config = model.peft_config[adapter_name]
  File "C:\Users\\Documents\GitHub\\venv\lib\site-packages\torch\nn\modules\module.py", line 1688, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'CLIPTextModel' object has no attribute 'peft_config'
```

### System Info

- `diffusers` version: 0.25.0.dev0
- Platform: Windows-10-10.0.22621-SP0
- Python version: 3.9.16
- PyTorch version (GPU?): 2.3.0.dev20240102+cu121 (True)
- Huggingface_hub version: 0.20.1
- Transformers version: 4.37.0.dev0
- Accelerate version: 0.23.0
- xFormers version: 0.0.21
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

**Comment by cerspense** [comment]
I am also experiencing this issue

**Comment by sayakpaul** [comment]
Did you install `peft`? 

Cc: @linoytsaban

**Comment by brandostrong** [comment]
> Did you install `peft`?
> 
> Cc: @linoytsaban

Yes, and the issue still persists.

**Comment by steverhoades** [comment]
Just ran into this issue as well using the script after 3 hours of training. Looks like it failed at the first checkpoint.

**Comment by linoytsaban** [comment]
I think the issue was that when using `train_text_encoder_ti` flag, we saved both text encoder and unet lora weights when using checkpointing when we should only save unet lora weights and the embeddings (since we're not modifying the text encoder weights when training with pivotal tuning). #6464 should fix that :)

**Comment by Heimdall-Nss** [comment]
I just find that this bug was fixed until version 0.29.2, but in the latest branch (0.30.0dev), specifically, after this commit https://github.com/huggingface/diffusers/commit/35f45ecd71a5c917406408a02bc982c3795d5a35, the problem reappeared.（Peft versions 0.7.0-0.11.0 cannot run properly）

**Comment by TornjV** [comment]
> I just find that this bug was fixed until version 0.29.2, but in the latest branch (0.30.0dev), specifically, after this commit [35f45ec](https://github.com/huggingface/diffusers/commit/35f45ecd71a5c917406408a02bc982c3795d5a35), the problem reappeared.（Peft versions 0.7.0-0.11.0 cannot run properly）

+1
